src/uucore/src/lib/features/quoting_style.rs

```diff
@@ -285,7 +293,15 @@ fn shell_with_escape(name: &str, quotes: Quotes) -> (String, bool) {
 }
 
 /// Escape a name according to the given quoting style.
-pub fn escape_name(name: &OsStr, style: &QuotingStyle) -> String {
+/// # Arguments
+/// * `name` - The name to be escaped.
+/// * `style` - The quoting style to use for escaping the name
+/// * `additional_escaped_chars` - A list of extra characters that should be treated as escaped chars for a specific context.
+pub fn escape_name(
+    name: &OsStr,
+    style: &QuotingStyle,
+    additional_escaped_chars: &[char],
+) -> String {
```

Comment:
Because `escape_name` is used in many places, and its `additional_escaped_chars` argument is acually used only once, I'd like to avoid having to add `&[]` to all the other function calls.

The way I did it in #6559 was just to rename the function to `escape_name_inner`, which actually take the parameter, and `escape_name` just calls `escape_name_inner` with `&[]`.

That way, you can create `escape_dirname` which calls `escape_name_inner` with `&[':']`.